mypyc/irbuild/util.py

```diff
@@ -17,21 +24,38 @@ def is_trait(cdef: ClassDef) -> bool:
     return any(is_trait_decorator(d) for d in cdef.decorators) or cdef.info.is_protocol
 
 
-def is_dataclass_decorator(d: Expression) -> bool:
-    return (
-        (isinstance(d, RefExpr) and d.fullname == 'dataclasses.dataclass')
-        or (
-            isinstance(d, CallExpr)
+def dataclass_decorator_type(d: Expression) -> Optional[str]:
+    if isinstance(d, RefExpr) and d.fullname in DATACLASS_DECORATORS:
+        return d.fullname.split('.')[0]
+    elif (isinstance(d, CallExpr)
             and isinstance(d.callee, RefExpr)
-            and d.callee.fullname == 'dataclasses.dataclass'
-        )
-    )
+            and d.callee.fullname in DATACLASS_DECORATORS):
+        name = d.callee.fullname.split('.')[0]
+        if name == 'attr' and 'auto_attribs' in d.arg_names:
+            auto = d.args[d.arg_names.index('auto_attribs')]
+            if isinstance(auto, NameExpr) and auto.name == 'True':
+                return 'attr-auto'
```

Comment:
Add comment saying that the mypy attrs plugin checks that the value of `auto_attribs` is not computed at runtime, so we don't need perform these checks here.

Reply:
done